sregistry/main/s3/pull.py

```diff
@@ -60,10 +60,10 @@ def pull(self, images, file_name=None, save=True, **kwargs):
             if e.response['Error']['Code'] == "404":
 
                 # Case 1, image not found, but not error with API
-                bot.error('Cannot find %s!' % name)      
+                bot.error('Cannot find %s!' % file_name)
 
                 # Try to help the user with suggestions
-                results = self._search_all(query=image)
+                results = self._search_all()
```

Comment:
Why are you removing the query?

Reply:
because the query argument isn't supported by the s3 implementation of search_all(). Was getting a TypeError...

Reply:
Most clients have a search that can support an arg (and if None or “” just search all). The solution is to fix to support that and not remove the argument entirely.

Reply:
Certainly, thought I had stumbled upon a bug :smile:

Reply:
Had another look at this and I really don't think it makes sense to add the `query` flag here. 
If this line in the code has been reached, the image doesn't exist in the bucket, so what should be done is just list all images that are present in the bucket, which is what `search_all` does.

The only `search_all` option that makes sense to me is `quiet`, which I think should be set to `True`, as currently the bucket images are printed twice.

Also, none of the other `search_all` methods have this kwarg.

Reply:
Here is an example of what I'm talking about: https://github.com/singularityhub/sregistry-cli/blob/master/sregistry/main/registry/query.py#L20 The general search function takes a query argument (typically the name of the image) and based on the formatting, will return a call to a sub-search function. In the case of sregistry (the link I showed above) we parse the query to determine if the user is searching for a collection or container, and if it's not, just return search_all() to list all. Since pull is given an image uri, we might consider just handing off the collection name (or some piece of the string) to search here https://github.com/singularityhub/sregistry-cli/blob/master/sregistry/main/s3/query.py#L18 and then returning a subset of results that might match. If you think it's more likely that buckets are small and listing all of the results is really the best option, we could consider that too.

Reply:
Oh I see, so basically all I need to do is use the `main.s3.search` method instead of `main.s3._search_all`?

Reply:
Yes, but if the user got the container name wrong, it might make sense to only search with the collection name. You could try a search with the collection name first, and if there are no results, instruct the user to use the `sregistry search` to list the available containers.

Reply:
Hmmm, would you mind fixing this yourself at some point? This would take more work than I am willing to put in on a friday afternoon :smile:

Reply:
Yes of course :)